### PR TITLE
Install provisioning plugins as well

### PR DESCRIPTION
**Pull request by lzap**
I want to test them on a daily basis as well

### CODE REVIEW

File: foreman_helper.bash [hunk 1]
```diff
@@ -35,3 +35,12 @@ tForemanGetTemplateId() {
   TPL_ID=$(hammer --csv template list --search "kind = \"$2\" AND name = \"$1\"" | tail -n1 | awk -F, '{print $1}')
 }
 
+tForemanInstallPlugin() {
+  for P in "$@"; do
```
Comment (ekohl) [inline foreman_helper.bash]: I'm somewhat allergic to short variables.

**Comment by ekohl** [comment]
I wonder if you'd want to install them through the installer, but this is a good start.

**Comment by lzap** [comment]
If @domcleal is not against, I'd love to have them in the basic installation test...

**Comment by mmoll** [comment]
I also would prefer to install the plugins with the installer as that also is testing a little more of the installer/puppet part, but adding a plugin after everything has already been running might also uncover some special ways to break.
:+1: in general anyway

**Comment by domcleal** [comment]
I agree that this should be done using the installer, not package installations.

I'd also prefer it as a separate test that we can enable, as I don't want to use this for our release tests, it could cause a lot of problems.  (Core shouldn't be dependent on plugins.)

The plugin list is also incomplete, it should use the list that the installer supports.

**Comment by domcleal** [comment]
I had a look at this earlier today, and this works to pull out and enable all plugins:

    foreman-installer --no-colors -v \
      $(foreman-installer --help | egrep -o -- "--(\[no-\])?enable-foreman-(proxy-)?plugin-\S+" | sed 's/\[no-\]//' | tr "\\n" ' ')

The trouble is that our installer options exist even if the plugin's not distributed on a given OS, so we'd need to exclude them.  Any thoughts about whether to put an exclusion list in foreman-bats (hrm) or could we add features to the installer to show plugin classes based on some condition?

**Comment by ares** [comment]
I'd prefer the second way. The easiest way how to do it is probably creating hook that would check which modules are used and based on facts it could print a warning and disable the module if it's not supported on this platform. Hiding the option completely could be done as well in a hook (boot hook I guess) but I think it would lead to misunderstandings like "why I don't see this option in same version of the installer".

**Comment by domcleal** [comment]
I have something roughly working as you describe, but am a little confused at kafo's hooks.  I could do this in a `pre` hook, but that's executed after the answers file is persisted, while `pre_validations` is executed before the interactive installer.  There doesn't seem to be a hook between the last user input and the parameters/modules being written.

Should I move pre up a little in kafo so it can still modify parameters, or add another hook - or am I missing something?

(https://github.com/theforeman/foreman-installer/compare/theforeman:develop...domcleal:plugin-oses)

**Comment by ares** [comment]
You're right, there's no built-in hook type for what you're trying to do. Luckily adding new hook type is as easy as [calling one method](https://github.com/theforeman/kafo/blob/master/lib/kafo/kafo_configure.rb#L56) and registering a type in [hooking.rb](https://github.com/theforeman/kafo/blob/master/lib/kafo/hooking.rb). You just have to find the right spot and name.

**Comment by domcleal** [comment]
Will do, thanks for the pointer!

**Comment by lzap** [comment]
Looks like Dom's pushing this into the installer. Thanks.

**Pull request closed without merging**